### ISSUE TITLE
Making PLKSR stable for real-world SISR

### ISSUE THREAD

**Issue by neosr-project** [dslisleedh/PLKSR#4]
Hi. First of all, thanks to everyone who participated on this research. Very thorough analysis on the paper.

As reported by others in [issue 3](https://github.com/dslisleedh/PLKSR/issues/3), PLKSR seems to be unstable for real-world SISR. GAN training is notoriously unstable, and causes issues even at lower learning rate.
So in an attempt to make it more stable, I have released a simple modification to PLKSR, named [RealPLKSR](https://github.com/muslll/neosr/blob/master/neosr/archs/realplksr_arch.py):
- Normalization was missing, as pointed by @dslisleedh. From my understanding, layer norm was avoided because of the impact on inference latency. I have tested multiple methods, including Instance norm, Layer norm, Batch norm, Group norm and RMSNorm. Because we usually train at lower batch sizes (<16), out of those tested, [Group Normalization](https://github.com/muslll/neosr/blob/master/neosr/archs/realplksr_arch.py#L87) performed best on my experiments. The impact on inference latency was minimal (~5% max). The number of groups was also tested. Increasing it leads to better regularization, but impacts convergence speed. The value `4` offered a good balance on all tests.
- Replacing GELU with [Mish](https://github.com/digantamisra98/Mish) on [channel mixer](https://github.com/muslll/neosr/blob/master/neosr/archs/realplksr_arch.py#L20). Mish showed better, more stable, convergence compared to GELU.
- Added [`nn.Dropout2d`](https://github.com/muslll/neosr/blob/master/neosr/archs/realplksr_arch.py#L130) to the last conv, as proposed in ["Reflash Dropout in Image Super-Resolution"](https://arxiv.org/pdf/2112.12089). Although not ideal, dropout is a simple method to increase generalization on real-world SISR.

Pretrained models:

| scale       | download                                                                                                                                 |
|--------------|-------------------------------------------------------------------------------------------------------------------------|
| 4x GAN   | [GDrive](https://drive.google.com/file/d/1iQmsnMhWXsHLYuYEwwRhdj9fMg71h-IU/view)   |
| 4x            | [GDrive](https://drive.google.com/file/d/12ek1vitEporWc5qqaYo6AMy0-RYlRqu8/view)       |
| 2x            | [GDrive](https://drive.google.com/file/d/1GAdf5VOqYa5ntswT9sYsKKZ2Z7OQp7gO/view) |

Training can be done on [neosr](https://github.com/muslll/neosr) using the following configurations: [paired dataset](https://github.com/muslll/neosr/blob/master/options/train_realplksr.toml) or through [realesrgan degradation pipeline](https://github.com/muslll/neosr/blob/master/options/train_realplksr_otf.toml).
Credits were acknowledged inside the [code](https://github.com/muslll/neosr/blob/master/neosr/archs/realplksr_arch.py#L103) and released under the same license as PLKSR (MIT). I hope this makes PLKSR more used under real-world degradations. It's a really impressive network. Thanks again for your research :+1:

**Comment by dslisleedh** [comment]
Thank you for your interest in this work, we are impressed with RealPLKSR's ability to stably learn real-world SISR tasks while maintaining low latency. We will add this issue and implementation to the readme so that many people can utilize your work!

**Comment by neosr-project** [comment]
Thanks @dslisleedh!

**Comment by Phhofm** [comment]
Just to add to this thread, I trained and released a RealPLKSR model on a dataset I degraded with a bit of lens blur, a bit of realistic noise, and a bit of jpg and webp (re)compression for photography.

The models and all the info to it can be found in its [Github Release here](https://github.com/Phhofm/models/releases/tag/4xNomosWebPhoto_RealPLKSR)

Some examples of my RealPLKSR model for visualization:
![334438914-763be319-fd46-4ca8-a4c2-e71e66b7cbc7](https://github.com/dslisleedh/PLKSR/assets/14755670/e40f5ade-15bc-4b12-82a9-cf420d804c3a)
![334438906-ebbb01d3-c0c7-427d-ba8b-a77797255f59](https://github.com/dslisleedh/PLKSR/assets/14755670/0c15b9a9-cf05-49e8-b96d-115b7c57a7d0)
![334438920-e713fde1-d5d2-4355-905f-9756482b5e6c](https://github.com/dslisleedh/PLKSR/assets/14755670/e9678be0-8568-430b-9bf4-9639277796da)
![334438924-624e054a-913a-431c-97a5-8406c5602151](https://github.com/dslisleedh/PLKSR/assets/14755670/208757bf-7ba7-40fc-9a9e-c240838e0430)
![334438928-42196634-c486-4b2b-adce-340f36af87fe](https://github.com/dslisleedh/PLKSR/assets/14755670/f4f38f06-78a5-4f4c-bd50-632ce61de6b3)
![334438931-d7d94828-7cdd-4e9d-9de3-497820899372](https://github.com/dslisleedh/PLKSR/assets/14755670/214df692-723d-44f2-a116-6cb49311f3da)
